velox/type/DecimalUtil.h

```diff
@@ -193,6 +194,48 @@ class DecimalUtil {
     return static_cast<TOutput>(rescaledValue);
   }
 
+  /// Rescale a double value to decimal value.
+  ///
+  /// Use `folly::tryTo` to convert a double value to int128_t or int64_t. It
+  /// returns an error when overflow occurs so that we could determine whether
+  /// an overflow occurs through checking the result.
+  ///
+  /// Normally, return the rescaled value. Otherwise, if the `toValue` overflows
+  /// the TOutput's limits or the `toValue` exceeds the precision's limits, it
+  /// will throw an exception.
+  template <typename TOutput>
+  inline static std::optional<TOutput> rescaleDouble(
+      double inputValue,
+      const int toPrecision,
```

Comment:
- drop 'const'
- rename: precision, scale ('to' is implied)
- rename: inputValue -> value